tests/test_process_domain_events.py

```diff
@@ -75,31 +80,19 @@ def test_not_going_to_infinite_loop_for_entities(empty_handler):
     aggregate.domain_events.register(domain_event)
     assert inspect.isfunction(empty_handler.empty_handle)
     assert domain_event in aggregate.domain_events
-    process_domain_events(lambda: aggregate.domain_events)
+    process_domain_events(lambda: take_domain_events(aggregate))
```

Comment:
it is better to add function instead of take_domain_events like so:
```
     def create_domain_events_collector(*aggregate_roots: AggregateRoot):
          return lambda: take_domain_events(aggregate)
```
(without using take_domain_events)